### DISCUSSION TITLE
Allow Email functionality to be completely disabled

### DISCUSSION THREAD

**Discussion by denolfe** [payloadcms/payload#1703] [Feature Requests & Ideas]
Currently, if an email transport isn't specified a mock email handler is used. We should provide a way to disable email functionality completely for testing and security purposes.

**Comment by iakshay** [comment]
Would be nice not to create mock credentials on every test run, currently I have to use `msw` to prevent API request going to `api.nodemailer.com` 😄

**Comment by wkillerud** [comment]
I have a scenario where this feature would have been useful. In my case I want to run Payload in a setting where it doesn't have direct access to the Internet. The database, media storage and server all run in this controlled network setting. For production I'll add an email transport to the setup, but for the dev/test environment at least I'm fine without it.

For me, Payload failed to start since the server didn't have network access to register the mock email.

I ended up patching in [this diff](https://github.com/payloadcms/payload/compare/master...wkillerud:payload:feat/disable-email?expand=1) locally, and it seems to be running OK from my limited testing on a fresh install.

If you'd like I can open a PR with the above diff and some documentation. Some feedback on what you'd like to see covered by tests would be helpful :)
